src/EnergyPlus/PluginManager.cc

```diff
@@ -1009,11 +1029,69 @@ namespace PluginManagement {
         char *strExcValue = (*EP_PyBytes_AsString)(pyStr2); // NOLINT(hicpp-signed-bitwise)
+        (*EP_Py_DECREF)(pyStr2);
         EnergyPlus::ShowContinueError("Python error description follows: ");
         EnergyPlus::ShowContinueError(strExcValue);
+
+        // See if we can get a full traceback.
+        // Calls into python, and does the same as capturing the exception in `e`
+        // then `print(traceback.format_exception(e.type, e.value, e.tb))`
+        PyObjectWrap pModuleName = (*EP_PyUnicode_DecodeFSDefault)("traceback");
+        PyObjectWrap pyth_module = (*EP_PyImport_Import)(pModuleName);
+        (*EP_Py_DECREF)(pModuleName);
+
+        if (pyth_module == NULL) {
+            EnergyPlus::ShowFatalError("Cannot find 'traceback' module in reportPythonError(), this is weird");
+            return;
+        }
+
+        PyObjectWrap pyth_func = (*EP_PyObject_GetAttrString)(pyth_module, "format_exception");
+        (*EP_Py_DECREF)(pyth_module); // PyImport_Import returns a new reference, decrement it
+
+        if (pyth_func || (*EP_PyCallable_Check)(pyth_func)) {
+
+            PyObjectWrap pyth_val = (*EP_PyObject_CallFunction3Args)(pyth_func, "OOO", exc_type, exc_value, exc_tb);
+
+            // traceback.format_exception returns a list, so iterate on that
+            if (!pyth_val || !(*EP_PyList_Check)(pyth_val)) { // NOLINT(hicpp-signed-bitwise)
+                EnergyPlus::ShowFatalError("In reportPythonError(), traceback.format_exception did not return a list.");
+            }
+
+            unsigned long numVals = (*EP_PyList_Size)(pyth_val);
+            if (numVals == 0) {
+                EnergyPlus::ShowFatalError("No traceback available");
+                return;
+            }
+
+            EnergyPlus::ShowContinueError("Python traceback follows: ");
+
+            EnergyPlus::ShowContinueError("```");
+
+            for (unsigned long itemNum = 0; itemNum < numVals; itemNum++) {
+                PyObjectWrap item = (*EP_PyList_GetItem)(pyth_val, itemNum);
+                if ((*EP_PyUnicode_Check)(item)) { // NOLINT(hicpp-signed-bitwise) -- something inside Python code causes warning
+                    std::string traceback_line = (*EP_PyUnicode_AsUTF8)(item);
+                    if (!traceback_line.empty() && traceback_line[traceback_line.length()-1] == '\n') {
+                        traceback_line.erase(traceback_line.length()-1);
+                    }
+                    EnergyPlus::ShowContinueError(" >>> " + traceback_line);
+                }
+                // PyList_GetItem returns a borrowed reference, do not decrement
+            }
+
+            EnergyPlus::ShowContinueError("```");
+
+            // PyList_Size returns a borrowed reference, do not decrement
+            (*EP_Py_DECREF)(pyth_val); // PyObject_CallFunction returns new reference, decrement
+        }
+        (*EP_Py_DECREF)(pyth_func); // PyObject_GetAttrString returns a new reference, decrement it
```

Comment:
New call from C to Python to get an actual traceback, then decode every line in said traceback and output it to the eplusout.err

Reply:
This is very cool.